bspacks/reason_bspack.sh

```diff
@@ -29,81 +29,152 @@ REFMT_PRE_CLOSURE="$buildDir/refmt_pre_closure"
-# disregard the usual entry point that is refmt_impl above (which takes care of
-# terminal flags parsing, etc.) and swap it with a new entry point, refmtJsApi (see below)
-../node_modules/bs-platform/bin/bspack.exe \
-  -bs-main Reason_toolchain \
-  -prelude-str "$resultStub" \
-  -I "$menhirSuggestedLib" \
-  -I "$reasonTargetDir/_build/default/src/ppx/"                               \
-  -I "$reasonTargetDir/_build/default/src/reason-merlin/"                     \
-  -I "$reasonTargetDir/_build/default/src/reason-parser/"                     \
-  -I "$reasonTargetDir/_build/default/src/reason-parser/vendor/easy_format/"  \
-  -I "$reasonTargetDir/_build/default/src/reason-parser/vendor/cmdliner/"     \
-  -I "$reasonTargetDir/_build/default/src/reason-parser-tests/"               \
-  -I "$reasonTargetDir/_build/default/src/refmt/"                             \
-  -I "$reasonTargetDir/_build/default/src/refmttype/"                         \
-  -I "$ocamlMigrateParseTreeTargetDir" \
-  -bs-MD \
-  -o "$REFMT_API.ml"
-
-# the `-no-alias-deps` flag is important. Not sure why...
-# remove warning 40 caused by ocaml-migrate-parsetree
-ocamlc -g -no-alias-deps -w -40 -I +compiler-libs ocamlcommon.cma "$REFMT_API.ml" -o "$REFMT_API.byte"
-# build REFMT_BINARY into an actual binary too. For testing purposes at the end
-ocamlc -g -no-alias-deps -w -40 -I +compiler-libs ocamlcommon.cma "$REFMT_BINARY.ml" -o "$REFMT_BINARY.byte"
-
-# compile refmtJsApi as the final entry file
-ocamlfind ocamlc -bin-annot -g -w -30 -w -40 -package js_of_ocaml,ocaml-migrate-parsetree -o "$REFMT_API_ENTRY" -I $buildDir -c -impl ./refmtJsApi.ml
-# link them together into the final bytecode
-ocamlfind ocamlc -linkpkg -package js_of_ocaml,ocaml-migrate-parsetree -g -o "$REFMT_API_FINAL.byte" "$REFMT_API.cmo" "$REFMT_API_ENTRY.cmo"
-# # use js_of_ocaml to take the compiled bytecode and turn it into a js file
-js_of_ocaml --source-map --debug-info --pretty --linkall +weak.js +toplevel.js --opt 3 --disable strict -o "$REFMT_PRE_CLOSURE.js" "$REFMT_API_FINAL.byte"
-# # use closure compiler to minify the final file!
-java -jar ./closure-compiler/closure-compiler-v20170910.jar --create_source_map "$REFMT_CLOSURE.map" --language_in ECMASCRIPT6 --compilation_level SIMPLE "$REFMT_PRE_CLOSURE.js" > "$REFMT_CLOSURE.js"
-
-# small integration test to check that the process went well
-# for the native binary
-echo "let f = (a) => 1" | "$REFMT_BINARY.byte" --parse re --print ml
-# for the js bundle
-node ./testRefmtJs.js
+
+build_reason_402 () {
+  # rebuild the project in case it was stale
+  cd ..
+  # We need 4.02 for js_of_ocaml (it has a stack overflow otherwise :/)
+  sed -i '' 's/"ocaml": "~4.6.0"/"ocaml": "~4.2.3004"/' ./esy.json
+  make pre_release
+  esy install
+  esy build
+  git checkout esy.json esy.lock
+  cd -
+}
+
+# Get OMP source from esy
+get_omp () {
+  mkdir $ocamlMigrateParseTreeTargetDir
+
+  esy build-env node_modules/@opam/ocaml-migrate-parsetree > _get_root
```

Comment:
In the latest esy you can make `esy.json` "scripts" entries which give you access to variables such as `#{ocaml-migrate-parsetree.build}`.